### PR TITLE
[oko] Clone additional repos from the correct branch

### PR DESCRIPTION
**Pull request by viroel**
Some repos are cloned during build and deploy step, but they are not considering which branch should be used. This patch fetch the proper info to clone and update associated steps.

### CODE REVIEW

**Comment by viroel** [comment]
We still need to set branches in install_yamls, and this would require this backport for dev-preview2: https://github.com/openstack-k8s-operators/install_yamls/pull/619

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-placement-operator-main-placement-operator-build-deploy-kuttl

**Comment by viroel** [comment]
Per logs[1]:

+ REF_BRANCH=main
+ export OPENSTACK_K8S_BRANCH=main
+ OPENSTACK_K8S_BRANCH=main

and 

+ git clone https://github.com/openstack-k8s-operators/install_yamls.git -b main

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/45523/rehearse-45523-pull-ci-openstack-k8s-operators-placement-operator-main-placement-operator-build-deploy-kuttl/1722983466284355584/artifacts/placement-operator-build-deploy-kuttl/openstack-k8s-operators-kuttl/build-log.txt

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh [hunk 1]
```diff
@@ -36,7 +38,7 @@ if [[ "$REF_ORG" != "$DEFAULT_ORG" ]]; then
     echo "Not a ${DEFAULT_ORG} job. Checking if isn't a rehearsal job..."
     EXTRA_REF_REPO=$(echo ${JOB_SPEC} | jq -r '.extra_refs[0].repo')
     EXTRA_REF_ORG=$(echo ${JOB_SPEC} | jq -r '.extra_refs[0].org')
-    #EXTRA_REF_BASE_REF=$(echo ${JOB_SPEC} | jq -r '.extra_refs[0].base_ref')
+    REF_BRANCH=$(echo ${JOB_SPEC} | jq -r '.extra_refs[0].base_ref')
```
Comment (marios) [inline ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh:41]: OK, i was going to ask why this was repeated but it is refs vs extra_refs

**Review by marios — APPROVED** [review]
looks right/didn't spot something and thanks for adding links to relevant logs from the test run

**Pull request merged**